### PR TITLE
Grant Summary Generation

### PR DESCRIPTION
**Pull request by StanHus**
Relation: https://eaflood.atlassian.net/jira/software/projects/FFAFGB/boards/1631?selectedIssue=FFAFGB-125
https://github.com/DEFRA/fcp-find-ai-data-ingester/pull/8

### CODE REVIEW

**Comment by StanHus** [comment]
@MohammedFoysal, I test on Sustainable Farming Incentive because thats the only thing I have so far in the summaries index search. Example:
Screenshot 2024-07-15 at 13 18 19

For now it sometimes spits out the wrong format (just text, no blocks) , is it an existing issue? Or did some of my changes cause it?

File: app/utils/langchain-utils.js [hunk 1]
```diff
@@ -63,8 +63,31 @@ const extractLinksForValidatingResponse = (jsonObj) => {
...
+
+    if (!summaries || summaries.length === 0) {
+      return []
+    }
+
+    // check if summaries are formatted as such: { title: string, scheme: string }
+    let error = false
+    summaries.forEach((summary) => {
+      if (!summary.title || !summary.scheme) {
+        error = true
```
Comment (MohammedFoysal) [inline app/utils/langchain-utils.js:78]: You can return early here instead of setting a flag. `return []`

**Review by MohammedFoysal — CHANGES_REQUESTED** [review]

**Comment by StanHus** [comment]
@MohammedFoysal , any ideas why it keeps producing the responses in a non-blocks kind of format. it just spits out text?

**Review by MohammedFoysal — APPROVED** [review]
Looks good. Don't forget to add summaries index name into ADO Library!

**Comment by StanHus** [comment]
@MohammedFoysal added, ready to merge

**Review by MohammedFoysal — APPROVED** [review]

**Pull request merged**